src/pyramid_retry/__init__.py

```diff
@@ -110,7 +107,8 @@ def retry_policy(environ, router):
             if number > 0:
                 # try to make sure this code stays in sync with pyramid's
                 # router which normally creates requests
-                request = router.make_request(environ)
+                request_ctx = router.request_context(environ)
+                request = request_ctx.begin()
```

Comment:
Oh, I see now. We re-use the original request unless we are on attempt 2, but by then the original context will have already been ended.

Yep, sorry for the noise.